zanata-war/src/main/java/org/zanata/action/ZanataRoleAction.java

```diff
@@ -0,0 +1,131 @@
+package org.zanata.action;
+
+import java.io.Serializable;
+import java.util.ArrayList;
+import java.util.List;
+
+import org.jboss.seam.annotations.Begin;
+import org.jboss.seam.annotations.Create;
+import org.jboss.seam.annotations.In;
+import org.jboss.seam.annotations.Install;
+import org.jboss.seam.annotations.Name;
+import org.jboss.seam.annotations.Scope;
+import org.jboss.seam.core.Conversation;
+import org.zanata.security.ZanataIdentity;
+import org.zanata.seam.security.ZanataJpaIdentityStore;
+import org.zanata.security.annotations.CheckLoggedIn;
+import org.zanata.security.annotations.CheckRole;
+import org.zanata.security.annotations.ZanataSecured;
+
+import static org.jboss.seam.ScopeType.CONVERSATION;
+import static org.jboss.seam.annotations.Install.APPLICATION;
+
+/**
+ * @author Patrick Huang
+ *         <a href="mailto:pahuang@redhat.com">pahuang@redhat.com</a>
+ */
+@Name("zanataRoleAction")
+@Scope(CONVERSATION)
+@Install(precedence = APPLICATION)
+@ZanataSecured
+@CheckLoggedIn
+@CheckRole("admin")
+public class ZanataRoleAction implements Serializable {
```

Comment:
On 2015-07-31 15:25, Patrick Huang wrote:
> In zanata-war/src/main/java/org/zanata/action/ZanataRoleAction.java
> <https://github.com/zanata/zanata-server/pull/908#discussion_r35947306>:
> 
>> +import org.zanata.security.annotations.ZanataSecured;
>> +
>> +import static org.jboss.seam.ScopeType.CONVERSATION;
>> +import static org.jboss.seam.annotations.Install.APPLICATION;
>> +
>> +/**
>> + * @author Patrick Huang
>> + *         <a href="mailto:pahuang@redhat.com">pahuang@redhat.com</a>
>> + */
>> +@Name("zanataRoleAction")
>> +@Scope(CONVERSATION)
>> +@Install(precedence = APPLICATION)
>> +@ZanataSecured
>> +@CheckLoggedIn
>> +@CheckRole("admin")
>> +public class ZanataRoleAction implements Serializable {
> 
> RoleAction is used by roledetail.xhtml and rolemanager.xhtml. They are
> both under admin/ which were restricted to admin only. Same reason for
> UserAction.

I don't think that current decision should be hard-coded at the level of
RoleAction and UserAction.  A lot of users who are currently admins
shouldn't be, so I can see these permissions being separated from admin
in future.


-- 
Sean Flanigan

Principal Software Engineer
Globalisation Tools Engineering
Red Hat